src/mdx/expandVersionedUrl.js

```diff
@@ -1,27 +1,71 @@
-const tokens = [
-  {
-    token: "$docs$",
-    rules: [
-      { match: "/optimize/", expandTo: "/docs/next" },
-      // these other mappings are currently wrong because there are no optimize versions yet...but eventually, this is what we'll want.
-      { match: "/optimize_versioned_docs/version-3.8.0/", expandTo: "/docs" },
-      {
-        match: "/optimize_versioned_docs/version-3.7.0/",
-        expandTo: "/docs/1.3",
-      },
-    ],
-  },
+// @ts-check
+
+/**
+ * @typedef {object} VersionMapping
+ * @property {string} docsVersion
+ * @property {string} optimizeVersion
+ */
+
+/** @type {Array<VersionMapping>} */
+const versionMappings = [
+  // When cutting a new version, add a new mapping here!
   {
-    token: "$optimize$",
-    rules: [
-      // TODO: introduce these mappings, as versions are added for optimize.
-      { match: "/docs/", expandTo: "/optimize/next" },
-      { match: "/versioned_docs/version-8.0/", expandTo: "/optimize" },
-      // { match: "/versioned_docs/version-1.3/", expandTo: "/optimize/3.7.0" },
-    ],
+    docsVersion: "8.0",
+    optimizeVersion: "3.8.0",
   },
+  { docsVersion: "1.3", optimizeVersion: "3.7.0" },
 ];
```

Comment:
This is the payoff of this PR. On the left side we would have had to align the source directories with URLs when adding a new version; on the right, all we have to do is add a new object that aligns the two versions.

Reply:
🤩 so CLEAN